src/lib/notifications/backgroundFetch.native.js

```diff
@@ -29,62 +32,90 @@ const log = logger.child({ from: 'backgroundFetch' })
-
-    const hasNewPayment = (type, status) => {
-      return type === 'receive' && status === 'completed'
+    if (taskId === MAIN_TASK) {
+      log.info('[BackgroundFetch] taskId: ', taskId)
+
+      const isLoggedIn = await AsyncStorage.getItem(IS_LOGGED_IN)
+      const oldNotifications = await AsyncStorage.getItem(OLD_NOTIFICATIONS)
+
+      log.info('isLoggedIn', isLoggedIn)
+
+      if (!isLoggedIn) {
+        return BackgroundFetch.finish(taskId)
+      }
+
+      const checkNotifications = async isInitialCall => {
+        try {
+          await hasConnection()
+          await onFeedReady(userStorage)
+        } catch (e) {
+          return BackgroundFetch.finish(taskId)
+        }
+
+        const lastFeedCheck = userStorage.userProperties.get('lastSeenFeedNotification')
+        const feed = await userStorage.getFeedPage(20, true)
+
+        log.info('lastFeedCheck', lastFeedCheck)
+        log.info('feed', feed)
+
+        const hasNewPayment = (type, status) => {
+          return type === 'receive' && status === 'completed'
+        }
+
+        const hasNewPaymentWithdraw = (type, status) => {
+          return type === 'send' && status === 'completed'
+        }
+
+        const newFeeds = feed.filter(feedItem => {
+          const { date, type, status } = feedItem
+          const feedDate = moment(new Date(date)).valueOf()
+          const isActual =
+            (hasNewPayment(type, status) || hasNewPaymentWithdraw(type, status)) &&
+            lastFeedCheck < feedDate &&
+            !includes(oldNotifications, feedItem.id)
+          return isActual && feedItem
+        })
+
+        userStorage.userProperties.safeSet('lastSeenFeedNotification', Date.now())
+
+        log.info('new feed items', { newFeeds })
+
+        log.info('pushing local notifications for feed items:', { total: newFeeds.length })
+
+        newFeeds.map(async feed => {
+          if (!isInitialCall) {
+            PushNotification.localNotification({
+              title: t`Payment from/to ${get(feed, 'data.counterPartyDisplayName', 'Unknown')} received/accepted`,
+              message: t`G$ ${get(feed, 'data.amount', 0)}`,
+              id: feed.id,
+              userInfo: { id: feed.id },
+            })
+          } else {
+            await AsyncStorage.setItem(OLD_NOTIFICATIONS, feed.id)
+          }
+        })
+      }
+
+      // Check notifications immediately on headless task call
+      await checkNotifications(true)
+
+      // Subscribe to feed update
+      // userStorage.feedStorage.feedEvents.on('updated', checkNotifications)
+    } else if (taskId === CLAIM_NOTIFICATION) {
+      const { entitlement: dailyUBI } = await goodWallet.getClaimScreenStatsFuse()
+      const notificationTimeout = new Date(dailyClaimTime.getTime() + bgFetchFrequency)
+      const needToNotify = dailyUBI && Date.now() > dailyClaimTime && Date.now() < notificationTimeout
+
+      const isNotified = await AsyncStorage.getItem(IS_NOTIFIED_OF_CLAIM)
```

Comment:
that means that if user didnt go to claim screen he will never see another notification.
if 24 hours passed since last notification he should be notified again.
ie he should be notified on any day he didnt claim yet.